Passing any alias type to interfaces.NewWithOptions causes panic in go@1.23

No issues with the `any` alias type with earlier versions of go, but had to upgrade to go@1.23.0 today and discovered that passing the `any` alias type to interfaces.NewWithOptions in 1.23.0 falls through to the default panic in `setFromType`: <https://github.com/rjeczalik/interfaces/blob/175268156d9381b6bcd86dd57846d3da482dba4f/type.go#L41-L73>

### Example output

```
go generate
panic: internal: t=*types.Alias, orig=*types.Slice

goroutine 1 [running]:
github.com/rjeczalik/interfaces.(*Type).setFromType(0xc0068d1840?, {0x759238?, 0xc000132140?}, 0x1?, {0x758520?, 0xc02d617d10?})
        [redacted]/github.com/rjeczalik/interfaces/type.go:73 +0x3fc
github.com/rjeczalik/interfaces.(*Type).setFromComposite(0xc0068d1840, {0x759488?, 0xc02d617d10?}, 0x0, {0x758520, 0xc02d617d10})
        [redacted]/github.com/rjeczalik/interfaces/type.go:119 +0xa9
github.com/rjeczalik/interfaces.(*Type).setFromType(0xc0068d1840, {0x758520?, 0xc02d617d10}, 0x0, {0x0?, 0x0?})
        [redacted]/github.com/rjeczalik/interfaces/type.go:69 +0x218
github.com/rjeczalik/interfaces.newType(...)
        [redacted]/github.com/rjeczalik/interfaces/type.go:32
github.com/rjeczalik/interfaces.buildInterfaceForPkg(0x6ab360?, 0xc0068d1e70)
        [redacted]/github.com/rjeczalik/interfaces/interface.go:144 +0x76a
github.com/rjeczalik/interfaces.buildInterface(0xc0068d1e70)
        [redacted]/github.com/rjeczalik/interfaces/interface.go:84 +0x1e9
github.com/rjeczalik/interfaces.NewWithOptions(0xc00017de70)
        [redacted]/github.com/rjeczalik/interfaces/interface.go:45 +0x125
```

Played around with this a bit and figured out a solution. PR to follow.

--- a/type.go
+++ b/type.go
@@ -65,6 +65,8 @@ func (typ *Type) setFromType(t types.Type, depth int, orig types.Type) {
 	case *types.Map:
 		typ.setFromComposite(t, depth, orig)
 		typ.setFromType(t.Key(), depth+1, orig)
+	case *types.Alias:
+		typ.setFromType(t.Underlying(), depth+1, orig)
 	case compositeType:
 		typ.setFromComposite(t, depth, orig)
 	default:

